From fe3464ca8710012a247bb4586dde21b080f88514 Mon Sep 17 00:00:00 2001
From: Jianyu Zhan <nasa4836@gmail.com>
Date: Sat, 19 Mar 2016 21:59:19 +0800
Subject: [PATCH] genirq: Remove redundant NULL check of irq_desc
Git-commit: fe3464ca8710012a247bb4586dde21b080f88514
Patch-mainline: v4.8-rc1
References: bsc#1019618

for_each_irq_desc() macro has already skipped NULL irq_desc, don't
bother to check it again.

Signed-off-by: Jianyu Zhan <nasa4836@gmail.com>
Cc: mingo@kernel.org
Cc: yhlu.kernel@gmail.com
Link: http://lkml.kernel.org/r/1458395959-7046-1-git-send-email-nasa4836@gmail.com
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Davidlohr Bueso <dbueso@suse.de>

---
 kernel/irq/proc.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/kernel/irq/proc.c b/kernel/irq/proc.c
index 4e1b94726818..50a8f28be247 100644
--- a/kernel/irq/proc.c
+++ b/kernel/irq/proc.c
@@ -421,12 +421,8 @@ void init_irq_proc(void)
 	/*
 	 * Create entries for all existing IRQs.
 	 */
-	for_each_irq_desc(irq, desc) {
-		if (!desc)
-			continue;
-
+	for_each_irq_desc(irq, desc)
 		register_irq_proc(irq, desc);
-	}
 }
 
 #ifdef CONFIG_GENERIC_IRQ_SHOW
-- 
2.6.6

